### PR TITLE
Draft overview for custom_expectations

### PR DESCRIPTION
**Pull request by abegong**
Changes proposed in this pull request:
- Adds an "Overview" section to the "creating custom expectations" part of the ToC
- Note that some content is unfinished, and will need to be either removed or completed before merging the `hacakthon-docs` branch into `develop`

@talagluck , can you please review for content?
@Rachel-Reverie  , can you please review for consistency with style guides and templates in other sections?

### CODE REVIEW

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (Rachel-Reverie) [inline docs/guides/expectations/creating_custom_expectations/overview.md]: "Has core logic exists and passes" reads as though a couple of variations of the statement were composed and part of the losing variation wasn't removed.  I'd remove the word 'exists.'

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (Rachel-Reverie) [inline docs/guides/expectations/creating_custom_expectations/overview.md]: I'm not certain this is accurate.  It seems to me that creating a custom expectation is one process, and contributing a created expectation to the open source is another process that is dependent on the creation process having been completed. Possibly you mean that creating a custom expectation and creating and contributing a custom expectation follow the same steps -- but that's still a bit inaccurate since the second process would have additional steps that the first stopped before reaching.

I would change this to: These processes compliment each other and their steps are streamlined so that one flows into the other.  Once you have created a custom expectation it is simple to contribute it to the open source project, and this section will teach you how to do both.

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (Rachel-Reverie) [inline docs/guides/expectations/creating_custom_expectations/overview.md]: "These code" -> "The code"

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (abegong) [reply docs/guides/expectations/creating_custom_expectations/overview.md]: ha. You're totally right.
```suggestion
* Has core logic that passes tests for all applicable `Execution Engines` and SQL dialects
```

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (abegong) [reply docs/guides/expectations/creating_custom_expectations/overview.md]:
```suggestion
These processes compliment each other and their steps are streamlined so that one flows into the other. Once you have created a custom Expectation, it is simple to contribute it to the open source project. This section will teach you how to do both.
```

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (abegong) [inline docs/guides/expectations/creating_custom_expectations/overview.md]:
```suggestion
The code to achieve the first four steps looks somewhat different depending on the class of Expectation you're developing. Accordingly, there are separate how-to guides and templates for each class of Expectation.
```

**Review by Rachel-Reverie — APPROVED** [review]

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (donaldheppner) [inline docs/guides/expectations/creating_custom_expectations/overview.md]: Could you link to: contributing/contributing for now?

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (donaldheppner) [inline docs/guides/expectations/creating_custom_expectations/overview.md:4]: It seems this could be a more broadly shared component rather than being local to create_custom_expectations

**Review by donaldheppner — CHANGES_REQUESTED** [review]

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (abegong) [reply docs/guides/expectations/creating_custom_expectations/overview.md:4]: Agreed. I actually copied it from `docs/guides/connecting_to_your_data/components/prerequisites.jsx`.

There are a few different versions of this already. AFACIT, they differ only in the defaults that they include. These defaults tend to differ by area within the ToC, so it makes sense to have them split out by subdirectory in the docs.

I'd suggest doing a separate PR to refactor all these `Prerequisites` components to use a common base class.

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (abegong) [reply docs/guides/expectations/creating_custom_expectations/overview.md]: Sure, although these docs won't go live until after another merge from `hackathon-docs` to `develop`.

Comment (abegong) [reply docs/guides/expectations/creating_custom_expectations/overview.md]:
```suggestion
You can find more detailed instructions in the [Contributing](/docs/contributing/contributing) section of the docs.
```

**Review by donaldheppner — APPROVED** [review]

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (talagluck) [inline docs/guides/expectations/creating_custom_expectations/overview.md:41]: Does it make sense to have a bullet for "Has thorough input validation and type checking, as determined by project code standards" ? It's a little vague, but I like the idea of encouraging more of this prior to elevating to production.

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (talagluck) [inline docs/guides/expectations/creating_custom_expectations/overview.md:8]: I think it should be "complement"

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (talagluck) [inline docs/guides/expectations/creating_custom_expectations/overview.md:48]: - "couple **of** tests"
- "diagonstics" is misspelled.

**Review by talagluck — COMMENTED** [review]
Love this! Just a few minor comments.

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (abegong) [reply docs/guides/expectations/creating_custom_expectations/overview.md:48]:
```suggestion
The docs in `Creating custom Expectations` focus on completing the four steps for experimental Expectations. Completing them will get to the point where your Expectation can be executed against one backend, with a couple of tests to verify correctness, and a basic docstring and metadata to support diagnostics. Optionally, you can also publish experimental Expectations to the [Great Expectations open source gallery](https://greatexpectations.io/expectations) by following the steps [here](overview#publishing-your-expectation-as-an-open-source-contribution).
```

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (abegong) [reply docs/guides/expectations/creating_custom_expectations/overview.md:8]:
```suggestion
These processes complement each other and their steps are streamlined so that one flows into the other. Once you have created a custom Expectation, it is simple to contribute it to the open source project. This section will teach you how to do both.
```

File: docs/guides/expectations/creating_custom_expectations/overview.md
Comment (abegong) [reply docs/guides/expectations/creating_custom_expectations/overview.md:41]: We have three other things that are similar in spirit:

* Has basic input validation and type checking (at the beta level)
* Passes all linting checks (production)
* Has passed a manual review by a code owner for code standards and style guides (production)

I figured these would cover the same substantive material. If you want to emphasize input validation and type checking, I suppose we could add a bullet, but it seems redundant to me.

**Pull request merged**
